dbt/models/eia860/_core_eia860__fgd_equipment/schema.yml

```diff
@@ -4,6 +4,22 @@ sources:
     tables:
       - name: _core_eia860__fgd_equipment
         data_tests:
+          - expect_columns_not_all_null:
+              exclude_columns:
+                - sorbent_type_3
+                - sorbent_type_4
+                - so2_equipment_type_4
+              conditional_columns:
+                county: EXTRACT(year FROM report_date) <= 2010
+                county_id_fips: EXTRACT(year FROM report_date) <= 2010
+                fgd_operating_date: EXTRACT(year FROM report_date) <= 2012
+                fgd_operational_status_code: EXTRACT(year FROM report_date) <= 2012
+                sludge_pond: EXTRACT(year FROM report_date) >= 2016
+                fgd_manufacturer: EXTRACT(year FROM report_date) <= 2012
+                fgd_manufacturer_code: EXTRACT(year FROM report_date) <= 2012
+                steam_plant_type_code: EXTRACT(year FROM report_date) >= 2014
+                plant_summer_capacity_mw: EXTRACT(year FROM report_date) <= 2010
+                water_source: EXTRACT(year FROM report_date) <= 2010
```

Comment:
In general, these conditions seem to take the form of "new column appeared in year X" or "old column was deprecated in year Y" -- the very few cases where there's a smattering of all null columns across a couple of years but not others I've added to the `exclude_columns` argument, because it seems like there are a few columns that are so rarely used that there are in fact many years in which there are zero values, even though they haven't been discontinued / added.

Alternatively, those could be specified explicitly with `IS IN (list, of, years)` -- which would be better?